### PR TITLE
Better Zoom

### PR DESCRIPTION
**Pull request by TheKodeToad**
- Scrolling
- Smoother animation
- (optional) Smooth camera

### CODE REVIEW

File: src/main/java/io/github/moehreag/axolotlclient/mixin/MinecraftClientMixin.java [hunk 1]
```diff
@@ -96,9 +98,21 @@ public void stop(CallbackInfo ci){
...
     public void tickClient(CallbackInfo ci){
         Color.tickChroma();
-
         AxolotlClient.TickClient();
         DiscordRPC.update();
+        Zoom.tick();
+    }
+
+
+    @Redirect(method = "tick", at = @At(value = "INVOKE", target = "Lorg/lwjgl/input/Mouse;getEventDWheel()I"))
```
Comment (moehreag) [inline src/main/java/io/github/moehreag/axolotlclient/mixin/MinecraftClientMixin.java]: I'd maybe put that in AxolotlClient.tickClient() (if using the tick method), but actually I'd rather go with a redirect mixin 
```suggestion
    @Redirect(method = "tick", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/player/PlayerInventory;method_3134(I)V"))
```
to only zoom further in while no screen is opened

**Review by moehreag — APPROVED** [review]
Looks good so far

File: src/main/java/io/github/moehreag/axolotlclient/mixin/MinecraftClientMixin.java (hunk 1 above)
Comment (TheKodeToad) [reply src/main/java/io/github/moehreag/axolotlclient/mixin/MinecraftClientMixin.java]: I looked at the code, and it's only executed while `currentScreen == null` anyway.

The redirect also allows for cancellation of the scroll event.

**Review by ghost — APPROVED** [review]
Smooth Camera is a great feature

**Pull request merged**